### PR TITLE
Spill to disk clean up

### PR DESCRIPTION
**Pull request by madsbk**
This PR clean up spilling to disk by introducing a `SpillToDiskProperties` class.

- [x] Fixing long file names (cc. @pentschev)
- [x] Clean up files on exit 
- [x] Fix race condition when multiple accesses to a proxy object overlap

### CODE REVIEW

**Review by pentschev — APPROVED** [review]
Thanks @madsbk for the hard work in putting this together. I can confirm this doesn't fail anymore with the sgkit workflows I've been working on. Although I must say this is still slower than the default spilling. To put this in perspective, the numbers I see are:

- JIT-Unspill (NFS): ~400 seconds;
- JIT-Unspill (local disk): ~75 seconds;
- Default spilling (regardless of local disk or NFS): ~38 seconds.

I'm not sure whether I'm missing something, but we can continue this discussion offline.

I have a couple minor typo fix suggestions and one question, but we're good to merge IMO.

File: dask_cuda/tests/test_proxify_host_file.py [hunk 1]
```diff
@@ -30,12 +31,24 @@
...
+if ProxifyHostFile._spill_to_disk is None:
+    # Hold on to `tmpdir` to keep dir alive until exit
+    tmpdir = tempfile.TemporaryDirectory()
+    ProxifyHostFile(
+        local_directory=tmpdir.name, device_memory_limit=1024, memory_limit=1024,
+    )
+assert ProxifyHostFile._spill_to_disk is not None
+
+# In order to use the same tmp dir, we use `root_dir` for all ProxifyHostFile creations
+root_dir = str(ProxifyHostFile._spill_to_disk.root_dir / ".." / "..")
```
[the remark this answers is not in the document]
Comment (madsbk) [reply dask_cuda/tests/test_proxify_host_file.py:47]: Added:
```python
# In order to use the same tmp dir, we use `root_dir` for all ProxifyHostFile creations
# Notice, we use `../..` to remove the `dask-worker-space/jit-unspill-disk-storage` part
# added by the ProxifyHostFile implicitly.
root_dir = str(ProxifyHostFile._spill_to_disk.root_dir / ".." / "..")
```

**Comment by madsbk** [comment]
@pentschev thanks for the review. Could I ask you to try reducing the data size to avoid spilling completely? I am wandering if the speed down you are seeing is because of `ProxyObject` overhead or something else.

**Comment by pentschev** [comment]
Failing tests are unrelated, reported in https://github.com/rapidsai/dask-cuda/issues/861 . I think @shwina may be able to look at this tomorrow, but may I ask you @madsbk to take a look in case this is something you can resolve quickly? FYI I asked @shwina to look at this since he wanted to be more familiarized with the spilling code, but of course this can be done in other ways/time too. 😄

**Comment by pentschev** [comment]
rerun tests

**Comment by madsbk** [comment]
@gpucibot merge

**Pull request merged**